server/graphql/asset/src/mutations/insert.rs

```diff
@@ -0,0 +1,214 @@
+use async_graphql::*;
+use chrono::NaiveDate;
+use graphql_core::{
+    simple_generic_errors::{
+        DatabaseError, InternalError, RecordAlreadyExist, UniqueValueViolation,
+    },
+    standard_graphql_error::{validate_auth, StandardGraphqlError},
+    ContextExt,
+};
+use service::{
+    asset::insert::{InsertAsset, InsertAssetError as ServiceError},
+    auth::{Resource, ResourceAccessRequest},
+};
+
+use crate::types::AssetNode;
+
+pub fn insert_asset(
+    ctx: &Context<'_>,
+    store_id: &str,
+    input: InsertAssetInput,
+) -> Result<InsertAssetResponse> {
+    let user = validate_auth(
+        ctx,
+        &ResourceAccessRequest {
+            resource: Resource::MutateAsset,
+            store_id: Some(store_id.to_string()),
+        },
+    )?;
+
+    let service_provider = ctx.service_provider();
+    let service_context = service_provider.context(store_id.to_string(), user.user_id)?;
+
+    match service_provider
+        .asset_service
+        .insert_asset(&service_context, input.into())
+    {
+        Ok(asset) => Ok(InsertAssetResponse::Response(AssetNode::from_domain(asset))),
+        Err(error) => Ok(InsertAssetResponse::Error(InsertAssetError {
+            error: map_error(error)?,
+        })),
+    }
+}
+
+#[derive(InputObject)]
+pub struct InsertAssetInput {
+    pub id: String,
+    pub store_id: Option<String>,
```

Comment:
I am wondering if you need store_id here given it will always be provided with the user?

Or maybe you do need it because a user from a certain store might want to add an asset to another store?

I'm not sure, just found it strange seeing two iterations of it on insert_asset in graphiql:
![Screenshot 2024-02-28 at 2 02 01 PM](https://github.com/msupply-foundation/open-msupply/assets/90807420/130218d1-e3fd-4abb-a613-923ea1e34788)

Reply:
Keen for @mark-prins opinion here. I agree if feels a bit weird!
I've done this because in theory according to the design docs you can create a asset that doesn't have a store assigned. So the on that's part of the input can be null, but you still need to send you logged in store so we can check permissions. I think it's probably ok how it is but probably needs some comments to explain...

Reply:
agree - weird looking but it makes some kind of sense. I'd keep it, and comment the weirdness